docs/src/userguide/fusedtypes.rst

```diff
@@ -162,24 +289,56 @@ by calling.
 Indexing
 --------
 
-You can index functions with types to get certain specializations, i.e.::
+You can index functions with types to get certain specializations, i.e.:
+
+.. tabs::
+
+    .. group-tab:: Pure Python
 
-    cfunc[cython.p_double](p1, p2)
+        .. literalinclude:: ../../examples/userguide/fusedtypes/indexing.py
+            :caption: indexing.py
 
-    # From Cython space
-    func[float, double](myfloat, mydouble)
+    .. group-tab:: Cython
 
-    # From Python space
-    func[cython.float, cython.double](myfloat, mydouble)
+        .. literalinclude:: ../../examples/userguide/fusedtypes/indexing.pyx
+            :caption: indexing.pyx
+
+Index functions can be called directly from python:
```

Comment:
Calling index functions from python is also discussed in __signatures__ section. It feels redundant. Should not merged together with this section? (This is the reason why I did not migrate that section to pure python)

Reply:
Yeah I agree with you here. `__signatures__` should probably be a very quick description of the attribute.

It seems like there are a few issues when indexing fused functions:
* When indexing from Python, or when indexing a `def` fused function from Cython, you must index with Python objects;
* `int` and `float` evaluate as the Python objects `int` and `float` don't let you select their equivalent C types;
* When indexing from Cython, `cython.double` (etc.) aren't available as runtime Python objects;
* When indexing from Python, `cython.double` (etc.) is only available if they have Cython installed, which may not be the case anyway!

The solution to all of this is to index with strings instead.